Makefile

```diff
@@ -32,9 +32,11 @@ generate: controller-gen ## Generate code and manifests
 
 test: test-unit test-e2e ## Run the tests
 
+ENVTEST_VERSION = $(shell go list -m k8s.io/client-go | cut -d" " -f2 | sed 's/^v0\.\([[:digit:]]\{1,\}\)\.[[:digit:]]\{1,\}$$/1.\1.x/')
 UNIT_TEST_DIRS=$(shell go list ./... | grep -v /test/)
 test-unit: ## Run the unit tests
-	$(Q)go test -count=1 -short $(UNIT_TEST_DIRS)
+	go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
+	eval $$(setup-envtest use -p env $(ENVTEST_VERSION)) && go test -count=1 -short $(UNIT_TEST_DIRS)
```

Comment:
Add envtest to the hack/tools/tools.go and use the pattern for installing this binary defined below?

Reply:
Yeah, that's fair.